Initialize config prior to content

sukrit007 opened the pull request:
[CADMIUM-269](https://jira.meltdev.com/browse/CADMIUM-269)
Initialize config prior to content so that operations like vault that does pre-processing of content has updated config prior to executing the content update.

**What issue are we facing ?**
If the content cache is removed, any site that relies on imsafe fails to fetch safety during startup as default config relies on gvault and config update happens after content.

**Steps to verify**
Deploy any site (with content folders removed).  During the deploy process, you should see config folder is checked our before content in content folder.

sukrit007 commented:
@jacobheun @BrendanErquiaga @FunnyGopher   ready for review. Let me know if you have any questions.

jacobheun commented:
@sukrit007 we should get some success and failure tests added around this to ensure the task sequencing is behaving properly.

sukrit007 commented:
@jacobheun  Good point. I might have to restructure the existing code though  to get unit tests going.  Unfortunately, this class does not have a way to inject ExecutorService so that I can introspect the thread pool. I will refactor the code and submit PR again. I do not have exceptional cases for sequencing so I might skip failure tests. Let me know if that works.

jacobheun commented:
@sukrit007 that should work, just want to ensure the standard workflows are covered.

sukrit007 commented:
@jacobheun  ready for review again. Added success test for sequencing. Added failure test as well (I am going to retain existing behavior for failure condition , but we might tweak it in future)
